crossbeam-channel/src/lib.rs

```diff
@@ -338,6 +331,13 @@
 //! [`Sender`]: struct.Sender.html
 //! [`Receiver`]: struct.Receiver.html
 
+#![doc(test(
+    no_crate_inject,
+    attr(
+        deny(warnings, rust_2018_idioms),
```

Comment:
`deny(warnings)` may not be forward compatible, as future versions of Rust may produce more warnings than before.  How about just leaving warnings as warnings and turning them into errors in CI?

Reply:
> `deny(warnings)` may not be forward compatible, as future versions of Rust may produce more warnings than before.

This `deny` only applies to the examples in the documentation and not to the published code, so there is no forward compatibility issue. If the compiler adds a new warning, it just breaks the CI.

> How about just leaving warnings as warnings and turning them into errors in CI?

We've looked into this before, but unfortunately couldn't find a way to apply deny to CI only: https://github.com/rust-lang/futures-rs/pull/1672#issuecomment-502018036